src/infra/marketplace/marketplaceQueryRepo.go

```diff
@@ -461,213 +456,334 @@ func (repo *MarketplaceQueryRepo) catalogItemFactory(
-	catalogItems, err := repo.ReadCatalogItems()
+	requestDto.Pagination = dto.Pagination{
+		PageNumber:   0,
+		ItemsPerPage: 1,
+	}
+	responseDto, err := repo.ReadCatalogItems(requestDto)
 	if err != nil {
 		return catalogItem, err
 	}
 
-	for _, catalogItem := range catalogItems {
-		for _, catalogItemSlug := range catalogItem.Slugs {
-			if catalogItemSlug.String() != slug.String() {
-				continue
-			}
-
-			return catalogItem, nil
-		}
+	if len(responseDto.MarketplaceCatalogItems) == 0 {
+		return catalogItem, errors.New("MarketplaceCatalogItemNotFound")
 	}
 
-	return catalogItem, errors.New("CatalogItemNotFound")
+	foundCatalogItem := responseDto.MarketplaceCatalogItems[0]
+	return foundCatalogItem, nil
 }
 
-func (repo *MarketplaceQueryRepo) ReadInstalledItems() (
-	[]entity.MarketplaceInstalledItem, error,
-) {
-	entities := []entity.MarketplaceInstalledItem{}
+func (repo *MarketplaceQueryRepo) ReadInstalledItems(
+	requestDto dto.ReadMarketplaceInstalledItemsRequest,
+) (responseDto dto.ReadMarketplaceInstalledItemsResponse, err error) {
+	model := dbModel.MarketplaceInstalledItem{}
+	if requestDto.MarketplaceInstalledItemId != nil {
+		model.ID = uint(requestDto.MarketplaceInstalledItemId.Uint16())
+	}
+	if requestDto.MarketplaceInstalledItemHostname != nil {
+		model.Hostname = requestDto.MarketplaceInstalledItemHostname.String()
+	}
+	if requestDto.MarketplaceInstalledItemType != nil {
+		model.Type = requestDto.MarketplaceInstalledItemType.String()
+	}
+	if requestDto.MarketplaceInstalledItemUuid != nil {
+		model.InstallUuid = requestDto.MarketplaceInstalledItemUuid.String()
+	}
+
+	dbQuery := repo.persistentDbSvc.Handler.
+		Where(&model).
+		Limit(int(requestDto.Pagination.ItemsPerPage))
+	if requestDto.Pagination.LastSeenId == nil {
+		offset := int(requestDto.Pagination.PageNumber) * int(requestDto.Pagination.ItemsPerPage)
+		dbQuery = dbQuery.Offset(offset)
+	} else {
+		dbQuery = dbQuery.Where("id > ?", requestDto.Pagination.LastSeenId.String())
+	}
+	if requestDto.Pagination.SortBy != nil {
+		orderStatement := requestDto.Pagination.SortBy.String()
+		orderStatement = strcase.ToSnake(orderStatement)
+		if orderStatement == "id" {
+			orderStatement = "ID"
+		}
+
+		if requestDto.Pagination.SortDirection != nil {
+			orderStatement += " " + requestDto.Pagination.SortDirection.String()
+		}
+
+		dbQuery = dbQuery.Order(orderStatement)
+	}
 
 	models := []dbModel.MarketplaceInstalledItem{}
-	err := repo.persistentDbSvc.Handler.
-		Model(&dbModel.MarketplaceInstalledItem{}).
-		Preload("Mappings").
-		Find(&models).Error
+	err = dbQuery.Preload("Mappings").Find(&models).Error
+	if err != nil {
+		return responseDto, errors.New("ReadMarketplaceInstalledItemsError")
+	}
+
+	var itemsTotal int64
+	err = dbQuery.Count(&itemsTotal).Error
 	if err != nil {
-		return entities, errors.New("ReadDatabaseEntriesError")
+		return responseDto, errors.New(
+			"CountMarketplaceInstalledItemsTotalError: " + err.Error(),
+		)
```

Comment:
Like I mentioned on the channel, the Limit can only be used AFTER the dbQuery.Count. Please refer to for the most updated version of this:
https://github.com/goinfinite/ez/blob/main/src/infra/containerQueryRepo.go#L309